Inc/HALALMock/Services/InputCapture/InputCapture.hpp

```diff
@@ -2,41 +2,40 @@
  * InputCapture.hpp
  *
  *  Created on: 30 oct. 2022
- *      Author: alejandro 
+ *      Author: alejandro
  */
 
 #pragma once
 #include "HALALMock/Models/PinModel/Pin.hpp"
 #include "HALALMock/Models/TimerPeripheral/TimerPeripheral.hpp"
+#include "HALALMock/Services/SharedMemory/SharedMemory.hpp"
 
 #ifdef HAL_TIM_MODULE_ENABLED
 
 class InputCapture {
-public:
-	class Instance {
-	public:
-		uint8_t id;
-		Pin pin;
-		TimerPeripheral* peripheral;
-		uint32_t channel_rising;
-		uint32_t channel_falling;
-		uint32_t frequency;
-		uint8_t duty_cycle;
+   public:
+    class Instance {
+       public:
+        uint8_t id;
+        Pin pin;
+        uint32_t* frequency;
+        uint8_t* duty_cycle;
+        bool is_active;
```

Comment:
didn't saw this until Jorge pointed it out, you are changing the public interface, you can add these variables as new members instead of changing the original because you don't know if the user will modify/access them. Return the implementation to how it was before and add these pointers as extra members